lib/HeroBonus.cpp

```diff
@@ -1318,10 +1309,20 @@ JsonNode Bonus::toJsonNode() const
 		root["valueType"].String() = vstd::findKey(bonusValueMap, valType);
 	if(stacking != "")
 		root["stacking"].String() = stacking;
+	if(description != "")
+		root["description"].String() = description;
+	if(effectRange != NO_LIMIT)
+		root["effectRange"].String() = vstd::findKey(bonusLimitEffect, effectRange);
+	if(duration != PERMANENT)
+		root["duration"].String() = vstd::findKey(bonusDurationMap, duration);

```

Comment:
This is wrong, duration is a bitmask

Reply:
Out of interest - if the duration set describes "when bonus is automatically removed", then what is the purpose of having `PERMANENT = 1` instead of `PERMANENT = 0`? If there is any ...

Reply:
duration is bitmask it can not have 0 value

Reply:
What I'm saying is that 0 = no bit set = not removal case since no bit set, which is logically the same as permanent duration.

Reply:
May be because it was not bitmask initially